events/apigw.go

```diff
@@ -42,6 +42,51 @@ type APIGatewayProxyRequestContext struct {
 	APIID         string                    `json:"apiId"` // The API Gateway rest API Id
 }
 
+// APIGatewayV2HTTPRequest contains data coming from the new HTTP API Gateway
+type APIGatewayV2HTTPRequest struct {
+	Version               string                         `json:"version"`
+	RouteKey              string                         `json:"routeKey"`
+	RawPath               string                         `json:"rawPath"`
+	RawQueryString        string                         `json:"rawQueryString"`
+	Cookies               []string                       `json:"cookies"`
+	Headers               map[string]string              `json:"headers"`
+	QueryStringParameters map[string]string              `json:"queryStringParameters"`
+	PathParameters        map[string]string              `json:"pathParameters"`
+	RequestContext        APIGatewayV2HTTPRequestContext `json:"requestContext"`
+	StageVariables        map[string]string              `json:"stageVariables"`
+	Body                  string                         `json:"body"`
+	IsBase64Encoded       bool                           `json:"isBase64Encoded,omitempty"`
+}
+
+// APIGatewayV2HTTPRequestContext contains the information to identify the AWS account and resources invoking the
+// Lambda function. It also includes Cognito identity information for the caller.
+type APIGatewayV2HTTPRequestContext struct {
+	RouteID      string                                    `json:"routeId"`
```

Comment:
It turns out `routeId` is a documentation error, and is not a part of the request model!

Reply:
Removed routeId